modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/impl/ParseExecutor.java

```diff
@@ -326,10 +327,13 @@ public XSDNamedComponent getDeclaration() {
         } else {
             //type is not simple, or complex with simple content, do a check 
             // for mixed
-            if (instance.getTypeDefinition() instanceof XSDComplexTypeDefinition
-                    && ((XSDComplexTypeDefinition) instance.getTypeDefinition()).isMixed()) {
-                //collape the text
-                text = Whitespace.COLLAPSE.preparse(text);
+            //TODO: If this came from a CDATA block we should not collapse it,
```

Comment:
This is a TODO, but still, it removes whitespace collapsing which is normally required (outside of CDATA)